src/components/tab-bar/partial-styles/tab-bar-scroller.scss

```diff
@@ -0,0 +1,86 @@
+.scroll-fade,
+.scroll-button {
+    position: absolute;
+    transition: {
+        property: transform;
+        duration: 0.3s;
+        delay: 0.5s;
```

Comment:
I like the delay for hiding the button, but I don't like the delay for showing the button. It makes it so that instead of something happening as a direct response to something I did, it instead appears to be acting "on its own".

Reply:
I don't have the same experience. I think the delay is so short, so I still experience it as a reaction to my actions, but with a slightly less rush than before. In my head, it's better to first see that the bar scrolled, then to see that the button appeared (as a result), than seeing everything happening in the same time.

Reply:
Really? I actually get slightly nauseous from it 😞

Reply:
Generally I don't think it's a good idea to animate many things in the UI simultaneously. Because users may miss what happened when too many things move around. (Unless they are gamers 😄)

It usually feels more peaceful and organic when there is slight delays and easing.
I hope this's not my personal opinion.


_However, what you say is possible to do in the CSS if the effect is too annoying._

Reply:
I think _any_ animation that isn't a _direct_ and _immediate_ response to my interaction is extremely attention-grabbing, and makes the application an absolute nightmare to work in. To me, such animations tell me "drop everything you're doing, forget your train of thought, and look at me _NOW_".

I'm already really annoyed with the animated open/close button on collapsible-sections.

If you really want to go that way, we need to implement support for `prefers-reduced-motion`.

Reply:
We can certainly work with the `prefers-reduced-motion`. But I don't know anything about how it should be used. In other words, how much should we _reduce_ animations? What I've seen on the web is that they simply remove animations or reduce fram rate. For me that's totally bizarre. Everything in our surrounding world is animated, and they have delays and easing and so forth. UI animations are suposed to mimic the real world physics and that is why they are used to add an additional layer of meaning to the interface. I have a hard time understanding how completely removing some animation helps.

Anyhow, this is something I have to read more about. We can start by looking at iOS's **Reduce Motion** and **Prefer cross-fade transitions**.

But for this particular issue. It's not a big deal for me to remove the delay for when things appear. So let me know if you want a fixup for that. ;)

Reply:
The reason for `prefers-reduced-motion` is that some people have disorders that make animations problematic. The [MDN page](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) mentions "vestibular motion disorders", which, after a quick search, seems to include what we'd colloquially would call "motion sickness" or "sea sickness".

For me, the problem is probably related to my ADHD, which makes it extremely difficult to filter out things that move or blink in my visual field. This means that it's difficult, and very tiring, for me to for example read a text when there are animations on screen. If I'm already tired or stressed, focusing on anything while such an animation is present becomes completely impossible.

None of our animations are ongoing, thankfully, but all animations that happen somewhere other than where my focus is grabs my attention. Every. Single. Time.

To me, animations like the ones on the open/close button for collapsible-section are annoying and tiring the same way I'm guessing it would be for you if we flashed a large popover with some annoying ad in it every time you hovered over any part of a collapsible-section header.

> But for this particular issue. It's not a big deal for me to remove the delay for when things appear. So let me know if you want a fixup for that. ;)

I would very much appreciate that 🙂

Reply:
Awesome. Thanks for the description Adrian. I'll do a fixup soon then.

It's so fascinating how different we all are and it's so good to have you onboard too, to get things more accessible for the type of challenges you and people like you face.

I sometimes feel the same about my dyslexia. It helps simplify things and reduce amount of attention grabbing stuff on the screen.

Reply:
Just tried it with the fixup, and I think it's great! Thank you!

Reply:
Resolved, but will leave as unresolved, because I don't want to hide the good discussion 😊